server/rest/src/main/java/org/infinispan/rest/InfinispanErrorResponse.java

```diff
@@ -14,7 +14,7 @@ protected InfinispanErrorResponse(Optional<InfinispanRequest> request) {
    }
 
    public static InfinispanErrorResponse asError(InfinispanRequest request, HttpResponseStatus status, String description) {
-      InfinispanErrorResponse infinispanResponse = new InfinispanErrorResponse(Optional.of(request));
+      InfinispanErrorResponse infinispanResponse = new InfinispanErrorResponse(Optional.ofNullable(request));
```

Comment:
This one is pretty interesting - when can we get a null request? I thought it's impossible.

Reply:
The incoming raw HttpRequest is never null, but it could be a bogus request that does not equate to anything useful (e.g., it's not a query request nor a cache api request), so this would lead to a null here, and Optional.of(null) leads to NPE